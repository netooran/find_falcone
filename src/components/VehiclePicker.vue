<template>
  <div class="vehicle-picker">
    <div class="vehicles">
      <Vehicle v-for="(vehicle, index) in vehicles" :vehicle="vehicle" :key="index"/>
    </div>
  </div>
</template>

<script>
import { mapState } from "vuex";
import Vehicle from "@/components/Vehicle.vue";

export default {
  name: "VehiclePicker",
  components: { Vehicle },
  computed: mapState(["vehicles"])
};
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped>
.vehicles {
  display: flex;
  justify-content: space-around;
  flex-wrap: wrap;
  margin: 4vw 8% 2vw 8%;
}
</style>
