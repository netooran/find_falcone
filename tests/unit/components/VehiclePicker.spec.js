import Vue from "vue";
import Vuex from "vuex";
import { shallowMount } from "@vue/test-utils";
import VehiclePicker from "@/components/VehiclePicker.vue";

Vue.use(Vuex);

describe("VehiclePicker", () => {
  let wrapper, vehicles;

  beforeEach(() => {
    vehicles = [
      { name: "Falcon Heavy", total_no: 2, speed: 2, max_distance: 437 },
      { name: "BFR", total_no: 2, speed: 2, max_distance: 437 }
    ];

    const store = new Vuex.Store({ state: { vehicles } });

    wrapper = shallowMount(VehiclePicker, { store });
  });

  it("has the correct value for vehicles", () => {
    expect(wrapper.vm.vehicles).toEqual(vehicles);
  });

  it("has the expected html structure", () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
