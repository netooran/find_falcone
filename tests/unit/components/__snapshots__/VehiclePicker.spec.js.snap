// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VehiclePicker has the expected html structure 1`] = `
<div
  class="vehicle-picker"
>
  <div
    class="vehicles"
  >
    <vehicle-stub
      vehicle="[object Object]"
    />
    <vehicle-stub
      vehicle="[object Object]"
    />
  </div>
</div>
`;
